### PR TITLE
Additional debugging of position IK solver, plus added virtual symbols

### PR DESCRIPTION
**Pull request by rethink-forrest**
* Some basic improvements to the position solver so that it stays within joint limits and stops if it gets stuck
* Added virtual symbol for proper inheritance
* Additional debug logging. Turn off later.

### CODE REVIEW

File: sns_ik_lib/include/sns_ik/osns_sm_velocity_ik.hpp [hunk 1]
```diff
@@ -37,7 +37,7 @@ class OSNS_sm_VelocityIK : public OSNSVelocityIK {
     OSNS_sm_VelocityIK(int dof, Scalar loop_period);
```
Comment (rethink-imcmahon) [inline sns_ik_lib/include/sns_ik/osns_sm_velocity_ik.hpp]: You need a `virtual` destructor here.

**Comment by rethink-imcmahon** [comment]
LGTM. Just add the `virtual` descructor and we'll be good to go.

**Comment by rethink-forrest** [comment]
I made the virtual destructor.

**Comment by rethink-imcmahon** [comment]
Great! Thanks

**Pull request merged**